### PR TITLE
Implement ReadStreamEvents operation

### PR DESCRIPTION
**Pull request by YoEight**

### CODE REVIEW

File: src/internal/types.rs [hunk 1]
```diff
@@ -276,3 +276,87 @@ impl TransactionId {
...
+    Error(String),
+    AccessDenied(String),
+}
+
+#[derive(Debug)]
+pub struct StreamSlice {
+    _from: i64,
+    _direction: ReadDirection,
+    _events: Vec<ResolvedEvent>,
+    _next_num_opt: Option<i64>,
```
Comment (Kerollmops) [inline src/internal/types.rs:315]: Why did you prefix fields with underscores ?

Comment (YoEight) [reply src/internal/types.rs:315]: I saw some Rust codebases where private fields were prefixed with an underscore. I don't know if I will maintain that guideline at the end.

Comment (Kerollmops) [reply src/internal/types.rs:315]: Please don’t 😊

Comment (Kerollmops) [reply src/internal/types.rs:315]: underscores are used to tell the compiler that you don't use variables/fields and avoid warning messages, same feature as Haskell. All struct fields are private by default, please don't follow Go's principle that uses capital letters to determine visibility.

**Pull request merged**